src/ty/mod.rs

```diff
@@ -26,6 +26,21 @@ pub enum Ty {
     Unit,
     /// The type of functions between two types.
     Arrow(Box<Ty>, Box<Ty>),
+    /// Type variable, used for unification.
+    Var(usize),
+}
+
+impl Ty {
+    /// Checks if a `Ty::Var` is contained inside the type.
+    ///
+    /// This method does not work with other `Ty` variants and must be used only with `Ty:Var`.
```

Comment:
Do we want an `assert` or `debug_assert` for this constraint?

Reply:
Hmmm... now that I think of `Ty::Var(x)` exposes `x` because this is an `enum` not an `struct`. Maybe we can change the method to take the `x` directly as its argument instead.